packages/admin/admin/src/router/ConfirmationDialog.tsx

```diff
@@ -1,74 +1,175 @@
+import { css, styled, useThemeProps } from "@mui/material/styles";
+import { ThemedComponentBaseProps } from "helpers/ThemedComponentBaseProps";
 import * as React from "react";
 import { FormattedMessage } from "react-intl";
 
 import { messages } from "../messages";
-import { RouterConfirmationDialogClassKey, styles } from "./ConfirmationDialog.styles";
 
+export type RouterConfirmationDialogClassKey =
+    | "root"
+    | "closeButton"
+    | "messageWrapper"
+    | "messageWarningIcon"
+    | "messageText"
+    | "actionButton"
+    | "saveButton"
+    | "discardButton";
+
+const StyledDialog = styled(Dialog, {
+    name: "CometAdminRouterConfirmationDialog",
+    slot: "root",
+    overridesResolver(_, styles) {
+        return [styles.root];
+    },
+})(css`
+    z-index: 1301;
+`);
+
+const CloseButton = styled(IconButton, {
+    name: "CometAdminRouterConfirmationDialog",
+    slot: "closeButton",
+    overridesResolver(_, styles) {
+        return [styles.closeButton];
+    },
+})(css`
+    position: absolute;
+    right: 14px;
+    top: 14px;
+    color: #fff;
+`);
+
+const MessageWrapper = styled("div", {
+    name: "CometAdminRouterConfirmationDialog",
+    slot: "messageWrapper",
+    overridesResolver(_, styles) {
+        return [styles.messageWrapper];
+    },
+})(css`
+    display: flex;
+`);
+
+const StyledWarning = styled(Warning, {
+    name: "CometAdminRouterConfirmationDialog",
+    slot: "messageWarningIcon",
+    overridesResolver(_, styles) {
+        return [styles.messageWarningIcon];
+    },
+})(css`
+    font-size: 20px;
+`);
+
+const MessageText = styled(Typography, {
+    name: "CometAdminRouterConfirmationDialog",
+    slot: "messageText",
+    overridesResolver(_, styles) {
+        return [styles.messageText];
+    },
+})(css`
+    padding-left: 10px;
+`);
+
+const SaveButton = styled(Button, {
+    name: "CometAdminRouterConfirmationDialog",
+    slot: "saveButton",
+    overridesResolver(_, styles) {
+        return [styles.actionButton, styles.saveButton];
+    },
+})(
+    ({ theme }) => css`
+        flex-grow: 1;
+        flex-basis: 50%;
+        margin-left: ${theme.spacing(2)};
+    `,
+);
+
+const DiscardButton = styled(Button, {
+    name: "CometAdminRouterConfirmationDialog",
+    slot: "discardButton",
+    overridesResolver(_, styles) {
+        return [styles.actionButton, styles.discardButton];
+    },
+})(
+    ({ theme }) => css`
+        flex-grow: 1;
+        flex-basis: 50%;
+        margin-right: ${theme.spacing(2)};
+    `,
+);
 export enum PromptAction {
     Cancel,
     Discard,
     Save,
 }
 
-export interface RouterConfirmationDialogProps {
+export interface RouterConfirmationDialogProps
+    extends ThemedComponentBaseProps<{
+        root: typeof Dialog;
+        closeButton: typeof IconButton;
+        messageWrapper: "div";
+        messageWarningIcon: typeof Warning;
+        messageText: typeof Typography;
+        actionButton: typeof Button;
```

Comment:
`actionButton` should not be necessary here, as the `slotProps` can already be set for `saveButton` and `discardButton` separately.